core/src/test/java/org/testcontainers/junit/FixedHostPortContainerTest.java

```diff
@@ -13,29 +22,76 @@
  * is started.
  */
 public class FixedHostPortContainerTest {
+    private static final String TEST_IMAGE = "alpine:3.2";
+
+    /**
+     * Default http server port (just something different from default)
+     */
+    private static final int TEST_PORT = 5678;
+
+    /**
+     * test response
+     */
+    private static final String TEST_RESPONSE = "test-response";
 
-    private static final int REDIS_PORT = 6379;
+    /**
+     * *nix pipe to fire test response on test port
+     */
+    private static final String HTTP_ECHO_CMD =
+        String.format("while true; do echo \"%s\" | nc -l -p %d; done", TEST_RESPONSE, TEST_PORT);
 
     @Test
     public void testFixedHostPortMapping() throws IOException {
         // first find a free port on the docker host that will work for testing
-        GenericContainer portDiscoveryRedis = new GenericContainer("redis:3.0.2").withExposedPorts(REDIS_PORT);
-        portDiscoveryRedis.start();
-        Integer freePort = portDiscoveryRedis.getMappedPort(REDIS_PORT);
-        portDiscoveryRedis.stop();
-
-
-        // Set up a FixedHostPortGenericContainer as if this were a @Rule
-        FixedHostPortGenericContainer redis = new FixedHostPortGenericContainer("redis:3.0.2").withFixedExposedPort(freePort, REDIS_PORT);
-        redis.start();
-
-//        Config redisConfig = new Config();
-//        redisConfig.useSingleServer().setAddress(redis.getContainerIpAddress() + ":" + freePort);
-//        Redisson redisson = Redisson.create(redisConfig);
-//
-//        redisson.getBucket("test").set("foo");
-//
-//        assertEquals("The bucket content was successfully set", "foo", redisson.getBucket("test").get());
-//        assertEquals("The container returns the fixed port from getMappedPort(...)", freePort, redis.getMappedPort(REDIS_PORT));
+        final Integer unusedHostPort;
+        String content;
+        try (final GenericContainer echoServer = new GenericContainer<>(TEST_IMAGE)
+            .withExposedPorts(TEST_PORT)
+            .withCommand("/bin/sh", "-c", HTTP_ECHO_CMD))
+        {
+            echoServer.start();
+
+            unusedHostPort = echoServer.getMappedPort(TEST_PORT);
+
+            content = this.readResponse(echoServer, unusedHostPort);
+
+            assertThat("Returned echo does not match expected", content, equalTo(TEST_RESPONSE));
+        }
+
+        // now starting echo server container mapped to known-as-free host port
+        try (final GenericContainer echoServer = new FixedHostPortGenericContainer(TEST_IMAGE)
```

Comment:
it would be better if you wrap it to the next line, somewhat like here:
https://github.com/testcontainers/testcontainers-java/blob/d372c3bf226a2286cf914750d26d3b90427771a0/core/src/test/java/org/testcontainers/junit/FileOperationsTest.java#L24